Mathlib/Data/Finset/Basic.lean

```diff
@@ -807,6 +807,11 @@ theorem ssubset_iff_exists_cons_subset : s ⊂ t ↔ ∃ (a : _) (h : a ∉ s),
   obtain ⟨a, hs, ht⟩ := not_subset.1 h.2
   exact ⟨a, ht, cons_subset.2 ⟨hs, h.subset⟩⟩
 
+theorem cons_swap (h : a ≠ b) (ha : a ∉ s) (hb : b ∉ s) :
```

Comment:
I think this will be annoying to rewrite with, since `hb` cannot be resolved by unification. Can you instead assume `hb : b ∉ s.cons a ha`?

Reply:
I think this also lets you drop `h`

Reply:
But `simp` will convert `b ∉ s.cons a ha` to `b ≠ a` and `b ∉ s`. Could you explain more about it? @eric-wieser

Reply:
If you write the lemma I'm asking for, then I can show you an example of where it succeeds and yours fails.

Reply:
I got your point. Rewriting is likely with `ha` and the new `hb`, not the three hypotheses which I have given.

Reply:
Here's the example:
```lean
import Mathlib

namespace Finset

variable {α} {s : Finset α}

theorem cons_swap_bad (h : a ≠ b) (ha : a ∉ s) (hb : b ∉ s) :
    (s.cons a ha).cons b (fun i ↦ hb ((mem_cons.mp i).elim (fun i ↦ False.elim (h (i ▸ rfl))) id))
    = (s.cons b hb).cons a (fun i ↦ ha ((mem_cons.mp i).elim (fun i ↦ False.elim (h i)) id)) := by
  apply Subset.antisymm <;> intro _ _ <;> simp [mem_cons] at * <;> tauto

theorem cons_swap_good (ha : a ∉ s) (hb : b ∉ s.cons a ha) :
    (s.cons a ha).cons b hb = (s.cons b fun h ↦ hb (mem_cons.mpr (.inr h))).cons a fun h ↦
      hb (mem_cons.mpr (.inl ((mem_cons.mp h).elim symm (fun h ↦ False.elim (ha h))))) := by
  apply Subset.antisymm <;> intro _ _ <;> simp [mem_cons] at * <;> tauto


example {x} (hx : x = cons 1 (cons 2 {3} (by simp)) (by simp)) :
    x = cons 2 (cons 1 {3} (by simp)) (by simp) := by
  rw [cons_swap_bad]
  · sorry -- creates two side goals
  · sorry
  · sorry

example {x} (hx : x = cons 1 (cons 2 {3} (by simp)) (by simp)) :
    x = cons 2 (cons 1 {3} (by simp)) (by simp) := by
  rw [cons_swap_good]
  sorry -- just the one goal
```